### ISSUE TITLE
MySQL Create DDL fails with Enum

### ISSUE THREAD

**Issue by mikebell90** [jOOQ/jOOQ#6461]
```java
 public void createBackfillStore() {
        DSLContext create = backFillDAO.getDSLContext();
        String sql = create
                .createTableIfNotExists(BackFillDAO.TABLE)
                .column(BackFillDAO.B_ID)
                .column(BackFillDAO.CREATED)
                .column(BackFillDAO.MODIFIED)
                .column(BackFillDAO.STARTED)
                .column(BackFillDAO.EVENT)
                .column(BackFillDAO.BEFORE)
                .column(BackFillDAO.AFTER)
                .constraints(
                        constraint("UN_BF").unique(BackFillDAO.EVENT),
                        constraint("PK_PF_ID").primaryKey(BackFillDAO.B_ID)
                ).getSQL();
        // work around jooq boog.
        sql = sql.replace("varchar ", "varchar(255) ");
        create.execute(sql);
```

This would have failed if it weren't for the replace statement. The reason is the fields before and and after are enums

```java
 static final Field<ExpectationType> BEFORE = field("bf",
            getEnumConverter(ExpectationType.class).nullable(false).length(255));
    static final Field<ExpectationType> AFTER = field("after",
            getEnumConverter(ExpectationType.class).nullable(false).length(255));
```

Note that even with the length(255) here, and the enum converter (below), the length is not generated, so a syntax error occurs. (just generates "Varchar" instead of "varchar(255)"

```java
public static <T extends Enum<T>> DataType<T> getEnumConverter(Class<T> clazz) {
        return SQLDataType.VARCHAR(255).asConvertedDataType(
                new EnumConverter<>(String.class, clazz));
    }
```

**Comment by lukaseder** [comment]
Thank you very much for your report. We'll look into this ASAP.

Related issue: #6241

**Comment by lukaseder** [comment]
This appears to be a duplicate of https://github.com/jOOQ/jOOQ/issues/6330, which was fixed in 3.10 and 3.9.4. What jOOQ version are you using?

**Comment by mikebell90** [comment]
Yes, it was 3.9.3, and 3.9.4 fixed it. Thanks
